### ISSUE TITLE
Uncaught TypeError: Cannot read property '$wire' of undefined

### ISSUE THREAD

**Issue by stewartmcfarland** [livewire/livewire#1727]
### Description
I'm using a Livewire hook `message.processed` in a script to retrieve the values of some component variables.

If I stay on the route where the component is used, everything works as expected. I can update values in a form, navigate forward and back (most variables are updating the queryString), interact with pagination, etc. and all is fine... the script can read the new values of the variables each time the component updates.

However, if I navigate away from the route (eg. click an item in the search results to view the details page, navigate to any other page on my site, or navigate to a completely different website)... then click Back to return to the site, all appears to be fine... however, if I click Back once more, the error is thrown. At that point the script can no longer retrieve the values of the variables, effectively rendering the script completely useless.

While my specific use case is quite complex, I was able to break it down into a very simple example. I also found that everything appears to work properly with Safari and the issue only occurs in Chrome and Firefox.
### Exact steps to reproduce

1. Visit the /test route.
2. Enter a value for the Text 1 input... values of `text1` and `text2` are logged in console.
3. Enter a value for the Text 2 input... values of `text1` and `text2` are logged in console.
4. Navigate the the /home route.
5. Click browser Back button... all looks fine.
6. Click browser Back button... error is thrown.
7. All subsequent updates to either text field, or forward/back navigation, will throw the error instead of logging the values in the console.

Note that once the error occurs, the rest of the component continues to function correctly as you navigate forward/back and/or modify the input fields again. However, anything that relies on the script being able to read the values of the variables will be broken.

### Stripped-down, copy-pastable code snippets
**Routes**
```
Route::get('/home', 'TestController@home');
Route::get('/livewire', 'TestController@livewire');
```
**TestController**
```
<?php

namespace App\Http\Controllers;

class TestController extends Controller
{
    public function home()
    {
        return view('testHome');
    }

    public function livewire()
    {
        return view('testLivewire');
    }
}
```
**testHome.blade.php**
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Test Home</title>
    <meta name="csrf-token" content="{{ csrf_token() }}">
    <livewire:styles />
</head>
<body>
    <h1>Home</h1>
    <h4><a href="/livewire">Livewire Test</a></h4>
    <livewire:scripts />
</body>
</html>
```
**testLivewire.blade.php**
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Test Livewire</title>
    <meta name="csrf-token" content="{{ csrf_token() }}">
    <livewire:styles />
</head>
<body>
    <h1>Livewire Test</h1>
    <livewire:test />
    <livewire:scripts />
</body>
</html>
```
**Test.php**
```<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Test extends Component
{
    public $text1;
    public $text2;

    protected $queryString = [
        'text1' => ['except' => ''],
        'text2' => ['except' => ''],
    ];

    public function render()
    {
        return view('livewire.test');
    }
}
```
**test.blade.php**
```
<div>
    <h4><a href="/home">Home</a></h4>
    <h4>Text 1: <input type="text" wire:model="text1"></h4>
    <h4>Text 2: <input type="text" wire:model="text2"></h4>

    <script>
        document.addEventListener("livewire:load", function(event) {
            Livewire.hook('message.processed', (message, component) => {
                var text1 = @this.get('text1');
                console.log('text1: ', text1);
                var text2 = @this.get('text2');
                console.log('text2: ', text2);
            });
        });
    </script>
</div>
```
**Context**
- Livewire version: 2.2.8
- Laravel version: 7.19
- Browser: Chrome, Firefox

**Comment by austenc** [comment]
@stewartmcfarland are you using Alpine or Turbolinks or anything like that?

**Comment by nuernbergerA** [comment]
You are using the wrong listener.

Sine v2 it's `DOMContentLoaded` instead of `livewire:load`.

 https://laravel-livewire.com/docs/2.x/lifecycle-hooks

**Comment by stewartmcfarland** [comment]
Thanks @nuernbergerA 

Hate it when I miss a simple change in the documentation. I noticed the changes to the hooks... but not the listener.

Using `livewire:load` I was able to retrieve the values I needed using plain Javascript instead of trying to use Livewire... so I wasn't pressured to find another fix.

I'll circle back to this at a later date when I have some time to test it using `DOMContentLoaded` as the listener and make sure all works as expected.
